src/adapter.ts

```diff
@@ -165,30 +196,26 @@ class MongooseAdapter {
    *
    * @param {Boolean} [commit=true] Flag that represents if automatic commit should be enabled or not
    */
-  setAutoCommit (commit = true) {
+  setAutoCommit(commit = true) {
     if (this.isSynced) this.autoCommit = commit;
   }
 
   /**
    * SyncedAdapter: Gets active session or starts a new one. Sessions are used to handle transactions.
    * @returns {Promise<Session>}
    */
-  async getSession () {
+  async getSession() {
     if (this.isSynced) {
-      return this.session && !this.session.hasEnded() ? this.session : this.mongoseInstance.startSession();
+      return this.session && this.session.inTransaction() ? this.session : this.mongoseInstance.startSession();
```

Comment:
I have been using these as of late, but I am not sure whether these methods were available in older mongodb versions or was it just because the documentation was so poor.

Reply:
@Zxilly

Reply:
@Sefriol both document and mongodb type definition told me to do this.
However, I'm really not sure about this. Currently I didn't have a MongoDB cluster on my dev machine. I will later deploy one.

Reply:
Yeah, I double checked this from their source and it was added 3 years ago. Should be fine.